packages/server/src/fhir/repo.ts

```diff
@@ -852,16 +858,22 @@ export class Repository extends BaseRepository implements FhirRepository<PoolCli
    * @param resourceType - The resource type.
    */
   async reindexResourceType(resourceType: ResourceType): Promise<void> {
+    const ctx = getAuthenticatedContext();
     if (!this.isSuperAdmin()) {
       throw new OperationOutcomeError(forbidden);
     }
+    ctx.logger.info('Reindex started', { resourceType });
 
-    await this.forAllResources(resourceType, async (resource) => {
-      try {
-        await this.reindexResourceImpl(resource);
-      } catch (err) {
-        getLogger().error('Failed to reindex resource', { error: normalizeErrorString(err) });
-      }
+    const startTime = Date.now();
+    let count = 0;
+    await this.forAllResources(resourceType, async (conn, resources) => {
+      await this.reindexResources(conn, ...resources);
```

Comment:
Regarding @codyebberson comments about the transaction in `forAllResources`, could the transaction be moved to wrap this line?

Reply:
That line is logically inside the transaction, with starts inside `forAllResources()` before the callback is invoked and also includes the search that returns `resources`